tests/Doctrine/Tests/ORM/Functional/Ticket/GH10348Test.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+final class GH10348Test extends OrmFunctionalTestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setUpEntitySchema([
+            GH10348Person::class,
+            GH10348Company::class,
+        ]);
+    }
+
+    public function testCanRemoveParentWithSelfReferencingChild(): void
+    {
+        $person1         = new GH10348Person();
+        $person2         = new GH10348Person();
+        $person2->mentor = $person1;
+
+        $company = new GH10348Company();
+        $company->addEmployee($person1)->addEmployee($person2);
+
+        $this->_em->persist($company);
+        $this->_em->flush();
+
+        $company = $this->_em->find(GH10348Company::class, $company->id);
+
+        $this->_em->remove($company);
+        $this->_em->flush();
+
+        self::assertEmpty($this->_em->createQuery('SELECT c FROM ' . GH10348Company::class . ' c')->getResult());
+        self::assertEmpty($this->_em->createQuery('SELECT p FROM ' . GH10348Person::class . ' p')->getResult());
```

Comment:
Are these assertions maybe superfluous and distracting? We're not testing the cascade here, are we? I think it would be best to remove them and use `@doesNotPerformAssertions`

Reply:
In another test that we merged recently Claudio suggested to have a check like this instead of `@doesNotPerformAssertions`. After all, the query returning an emtpy result is the best indication of things actually happening (entities being deleted).

I'm indifferent – you decide.